### PR TITLE
get_publishers()

### PR DESCRIPTION
**Pull request by nimshi89**
adding the `get_publishers()` function

### CODE REVIEW

File: src/store/metadata/oxigraph/store.py [hunk 1]
```diff
@@ -258,7 +259,40 @@ def get_publishers(self) -> Optional[Dict]:  # pragma: no cover
...
+        )
+
+        # Serialize the graph into jsonld
+        data = json.loads(result.serialize(format="json-ld"))
+
+        # Use a context file to shape our jsonld, removing long form references
+        data = jsonld.flatten(
+            data, {"@context": constants.CONTEXT, "@type": "hydra:Collection"}
+        )
+
```
Comment (SarahJohnsonONS) [inline src/store/metadata/oxigraph/store.py:281]: Add a comment here so we know to go back and remove this when the context weirdness is resolved

File: src/store/metadata/oxigraph/store.py [hunk 2]
```diff
@@ -258,7 +259,40 @@ def get_publishers(self) -> Optional[Dict]:  # pragma: no cover
...
+        # TODO Fix context weirdness - at the moment, the flatten() method is changing @type to `versions_url`
+        # also adds dcat: to the publishers
+        data["@graph"][0]["@type"] = "hydra:Collection"
+        data["@graph"][0]["publishers"] = data["@graph"][0].pop("dcat:publisher")
+
+        for idx, publisher in enumerate(data["@graph"][0]["publishers"]):
```
Comment (SarahJohnsonONS) [inline src/store/metadata/oxigraph/store.py:287]: Mike said try and avoid using `enumerate` if you can use a comprehension instead, e.g. this works as well:
```
    data["@graph"][0]["publishers"] = [
        self.get_publisher(x["@id"].split("/")[-1])
        for x in data["@graph"]
        if "landing_page" in x.keys()
    ]
    del data["@graph"][0]["dcat:publisher"]
```

File: src/store/metadata/oxigraph/store.py [hunk 3]
```diff
@@ -258,7 +259,40 @@ def get_publishers(self) -> Optional[Dict]:  # pragma: no cover
...
+            publisher_id = publisher["@id"].split("/")[-1]
+            data["@graph"][0]["publishers"][idx] = self.get_publisher(publisher_id)
+        
+        # TODO Update @context so it's not hardcoded
+        data["@graph"][0]["@context"] = "https://staging.idpd.uk/ns#"
+        result = data["@graph"][0]
```
Comment (SarahJohnsonONS) [inline src/store/metadata/oxigraph/store.py:293]: `@context` is still missing so the test is failing on validation

**Review by SarahJohnsonONS — COMMENTED** [review]
Comments added

File: src/store/metadata/oxigraph/store.py (hunk 3 above)
Comment (SarahJohnsonONS) [reply src/store/metadata/oxigraph/store.py:293]: Weird, this wasn't here when i added the comment

Comment (nimshi89) [reply src/store/metadata/oxigraph/store.py:295]: Yeah it is odd, I don't see why this issue remains but for now it solves it

**Pull request merged**